libobs/util/platform-nix.c

```diff
@@ -96,10 +102,56 @@ void os_dlclose(void *module)
 		dlclose(module);
 }
 
+#if !defined(__APPLE__) && OBS_QT_VERSION == 6
+int module_has_qt5_check(const char *path)
+{
+	void *mod = os_dlopen(path);
+	if (mod == NULL) {
+		return 1;
+	}
+
+	struct link_map *list = NULL;
+	if (dlinfo(mod, RTLD_DI_LINKMAP, &list) == 0) {
+		for (struct link_map *ptr = list; ptr; ptr = ptr->l_next) {
+			char *slash = strrchr(ptr->l_name, '/');
+			if (!slash)
+				continue;
+			char *base = slash + 1;
+			if (strncmp(base, "libQt5", 6) == 0) {
+				return 0;
+			}
```

Comment:
I wonder if it wouldn't be simpler to

```c
if (strstr(ptr->l_name, "libQt5") != NULL)
	return 0;
```

here.

(There might be a concern that a well crafted LD_PRELOAD with a maliciously crafted library path would trick strstr() into an out-of-bounds read and potentially arbitrary code execution, but given this is a subprocess and this is purely hypothetical, I don't think it matters in practice.)

Reply:
We could add a strlen to avoid reading past the end of the string. I don't really feel stongly since I assume no one is insane enough to put their libraries in a folder called `libQt5`. (if you are the person filling the bug report because your install in the libQt5 folder is broken, sorry).